tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -61,31 +59,30 @@ public static AIBotConfig fromMailetConfig(MailetConfig mailetConfig) throws Mai
         if (Strings.isNullOrEmpty(gptAddressParam)) {
             throw new MailetException("No value for " + GPT_ADDRESS_PARAMETER_NAME + " parameter was provided.");
         }
-        
-        MailAddress mailAddress;
+
+        Optional<URL> baseURLOpt = toOptionalIfNotEmpty(baseUrlParam)
+                .flatMap(AIBotConfig::baseURLStringToURL);
+
+        LlmModel llmModel = !Strings.isNullOrEmpty(llmModelParam)
+                            ? parseLlmModelParamOrThrow(llmModelParam)
+                            : DEFAULT_LLM_MODEL;
```

Comment:
Use optionals in order to avoid conditions?

Reply:
I wasn't aware of the `Optional.ofNullable(...).filter(...)` trick, that's really helpful!